lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1312,10 +1312,14 @@ sub _failure_reason {
     my @failed_modules;
     my $modules = $self->modules;
     while (my $m = $modules->next) {
-        next if ($m->result eq INCOMPLETE);
-        push(@failed_modules, $m->name . ':' . $m->result);
+        if ($m->result eq FAILED || $m->result eq SOFTFAILED) {
+            push(@failed_modules, $m->name . ':' . $m->result);
+        }
+        if ($m->result eq FAILED && $m->fatal) {
+            return $m->name;
+        }
     }
-    return join('', @failed_modules) || $self->result;
+    return join(',', sort (@failed_modules)) || $self->result;
```

Comment:
Hmm.. instead of calling `sort` on string values I prefer to query database for ordered rows. I think something like `my $modules = $self->modules({order_by => 'id'});` or `my $modules = $self->modules({order_by => 't_created'});`